### PR TITLE
Cloud federation credential abuse

### PR DESCRIPTION
**Pull request by rosplk**
Updated ymls to make sure validate passes and other minor updates

TO DO: Review/rewrite search logic and automated testing

### CODE REVIEW

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

AWS SAML Update identity provider :white_check_mark:
- diskUsage: 131072
- runDuration: 0.748s
- scanCount: 10

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

AWS SAML Access by Provider User and Principal :white_check_mark:
- diskUsage: 131072
- runDuration: 0.74s
- scanCount: 28

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

O365 Add App Role Assignment Grant User :white_check_mark:
- diskUsage: 131072
- runDuration: 0.749s
- scanCount: 24

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

O365 Excessive SSO logon errors :white_check_mark:
- diskUsage: 122880
- runDuration: 0.771s
- scanCount: 23

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

O365 Added Service Principal :white_check_mark:
- diskUsage: 135168
- runDuration: 0.765s
- scanCount: 6

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

O365 New Federated Domain Added :x:
- diskUsage: 122880
- runDuration: 0.672s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

O365 New Federated Domain Added :x:
- diskUsage: 122880
- runDuration: 0.649s
- scanCount: 0

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

O365 New Federated Domain Added :white_check_mark:
- diskUsage: 126976
- runDuration: 0.778s
- scanCount: 2

**Review by P4T12ICK — DISMISSED** [review]
Great work Rod

File: detections/cloud/aws_saml_access_by_provider_user_and_principal.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+author: Rod Soto, Splunk
+date: '2021-01-26'
+description: This search provides specific SAML access from specific Service Provider,
+  user and targeted principal at AWS. This search provides specific information to
```
Comment (miskoSplunk) [inline detections/cloud/aws_saml_access_by_provider_user_and_principal.yml:4]: @rosplk could you please clarify the context of a principal - is it a "windows" term, they use authentication/security principles as an umbrella term for anything that gets authenticated?

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :x:
- diskUsage: 483328
- runDuration: 1.2610000000000001s
- scanCount: 3

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :x:
- diskUsage: 483328
- runDuration: 1.3800000000000001s
- scanCount: 3

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :x:
- diskUsage: 483328
- runDuration: 1.37s
- scanCount: 2

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :x:
- diskUsage: 483328
- runDuration: 1.339s
- scanCount: 2

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :x:
- diskUsage: 483328
- runDuration: 1.337s
- scanCount: 2

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :x:
- diskUsage: 483328
- runDuration: 1.373s
- scanCount: 2

**Comment by patel-bhavin** [comment]
test is now running locally using attack_range : `python attack_range.py test -tf ../security_content/tests/endpoint/certutil_exe_certificate_extraction.test.yml`

image


failing while using AWS Batch

**Comment by P4T12ICK** [comment]
This PR was created by Automated Detection Testing Service :robot:

Tested detections:

Certutil exe certificate extraction :white_check_mark:
- diskUsage: 487424
- runDuration: 1.329s
- scanCount: 13

**Review by patel-bhavin — APPROVED** [review]

**Comment by patel-bhavin** [comment]
All tests have passed! awesome work @rosplk @P4T12ICK

**Pull request merged**